build-logic/jvm/src/main/kotlin/gradlebuild/jvm/extension/UnitTestAndCompileExtension.kt

```diff
@@ -51,6 +53,9 @@ abstract class UnitTestAndCompileExtension(
     private
     fun enforceCompatibility(majorVersion: Int) {
         tasks.withType<JavaCompile>().configureEach {
+            javaCompiler.set(project.javaToolchains.compilerFor {
```

Comment:
❌ Let's not import this "internal" accessor but use the API instead
```suggestion
            javaCompiler.set(project.the<JavaToolchainService>().compilerFor {
```

This would require an import for `org.gradle.jvm.toolchain.JavaToolchainService`.

Reply:
Should the [documentation](https://docs.gradle.org/current/userguide/toolchains.html#toolchains_for_tasks) be updated then? I've got the idea from there.

Reply:
Documentation shows a script, we are not in a script so we don't have accessors
See https://docs.gradle.org/current/userguide/kotlin_dsl.html#type-safe-accessors